### PR TITLE
Add auto-compound demo

### PR DESCRIPTION
**Pull request by imstar15**
### Screenshoot:
image

### Log:
```
SENDER_MNEMONIC="<SENDER_MNEMONIC>" PROVIDER_URL="ws://127.0.0.1:9946" node src/scheduleDynamicDispatchTaskToAutoCompound.js
2023-11-21 15:38:06        API/INIT: RPC methods not decorated: automationPrice_queryFeeDetails
Unable to map [u8; 32] to a lookup index
2023-11-21 15:38:06        API/INIT: turing/295: Not decorating unknown runtime apis: XcmpHandlerApi/1, AutomationTimeApi/1, AutomationPriceApi/1

1. Getting system properties...

2. Checking account free balance...
The account free balance: 1000 TUR

3. Delegating to collator...
delegatorState:  null
delegatorWalletAddress:  67RKQFDRWRFnY7RrkvR7N1kNpZiXa9P2qdLtwxYZFDjs4dAr
a. Minimum Amount to be staked: 500000000000
b. Candidate Delegation Count: 0
c. Delegation Count: 0
d. autocompound Delegation Count: 0
status.type:  Ready
Unable to map [u8; 32] to a lookup index
status.type:  InBlock
status.type:  Finalized
Delegate, extrinsicHash: 0x4ffe2cdaf8acc0e18db247146deb9b225dccd01c06089229e88886da0e255416, blockHash: 0xc9572db6933feeca17ac276c0562c3036b97d8c65ec6d89176c8400702a67c13

4. Scheduling a dynamic task to auto compound...
status.type:  Ready
Unable to map [u8; 32] to a lookup index
status.type:  InBlock
status.type:  Finalized
extrinsicHash: 0x7b3e6886608e8e91f50673375bf569c38541446501b23ee55fe866ecc90d723e, blockHash: 0xf890dc8f0b20c0a7a8a5519b3d26805ac923014a5bf93dc0a01b5376fb9a2ed4,
Found the automationTime.TaskScheduled event and retrieved TaskId, 5-2-3

5. Listening to the parachainStaking.DelegationIncreased event...
        parachainStaking:DelegationIncreased:: (phase=Initialization)
                        AccountId32: 67RKQFDRWRFnY7RrkvR7N1kNpZiXa9P2qdLtwxYZFDjs4dAr
                        AccountId32: 69vaEyuq6gN3nJQ3BZGqkiNLXV5r8ub3HBqtgudrfQbMnzxx
                        u128: 10000000000
                        bool: true
Found the parachainStaking.DelegationIncreased event {
  phase: 'Initialization',
  event: {
    method: 'DelegationIncreased',
    section: 'parachainStaking',
    index: '0x160d',
    data: {
      delegator: '67RKQFDRWRFnY7RrkvR7N1kNpZiXa9P2qdLtwxYZFDjs4dAr',
      candidate: '69vaEyuq6gN3nJQ3BZGqkiNLXV5r8ub3HBqtgudrfQbMnzxx',
      amount: '10,000,000,000',
      inTop: true
    }
  },
  topics: []
}
```

### CODE REVIEW

File: demo/src/scheduleDynamicDispatchTaskToAutoCompound.js [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  const provider = new WsProvider(providerUrl);
+  const api = await ApiPromise.create({ provider, rpc, types });
+
+  // Get system properties
+  console.log("\n1. Getting system properties...");
+  const systemProperties = await api.rpc.system.properties();
+  const tokenDecimals = systemProperties.tokenDecimals.unwrap()[0];
+  const tokenDecimalsNumber = new BN(10).pow(new BN(tokenDecimals.toNumber()));
+  const tokenSymbol = systemProperties.tokenSymbol.unwrap()[0].toString();
+  const ss58Format = systemProperties.ss58Format.unwrap().toNumber();
```
Comment (chrisli30) [inline demo/src/scheduleDynamicDispatchTaskToAutoCompound.js:176]: This is a good idea to retrive decimals and ss58 from RPC, and needs to go into adapter as a function, `getMetadata()`

File: demo/src/scheduleDynamicDispatchTaskToAutoCompound.js [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+  const keyringPair = await getKeyringPair(ss58Format);
+
+  // Read account free balance
+  console.log("\n2. Checking account free balance...");
```
Comment (chrisli30) [inline demo/src/scheduleDynamicDispatchTaskToAutoCompound.js:181]: TODO: try to wrap up the auto-staking logic in adapter

adapter.bondMoreAutoCompound(collator, schedule{type, startTime, interval}, amount);

File: demo/src/scheduleDynamicDispatchTaskToAutoCompound.js [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+            events,
+            foundEvent,
+            foundEventIndex,
+          });
+        }
+      });
+    };
+
+    listenSystemEvents().catch(console.log);
+  });
```
[the remark this answers is not in the document]
Comment (imstar15) [reply demo/src/scheduleDynamicDispatchTaskToAutoCompound.js]: Fixed

**Review by chrisli30 — APPROVED** [review]
Okay, looks good. Good to merge!

**Pull request merged**